### PR TITLE
support missing role-arn and role-session-name

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license] (https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request, I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
support missing role-arn and role-session-name

## Motivation and Context
to complete the AWS IAM support additional ENVs
are needed.

## How to test this PR?
The PR is simply allows ROLE_ARN

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

**Review by pjuarezd — APPROVED** [review]

File: cmd/client-s3.go [hunk 1]
```diff
@@ -219,6 +220,12 @@ func getCredentialsChainForConfig(config *Config, transport http.RoundTripper) (
 		// set AWS_WEB_IDENTITY_TOKEN_FILE is MC_WEB_IDENTITY_TOKEN_FILE is set
 		if val := env.Get("MC_WEB_IDENTITY_TOKEN_FILE", ""); val != "" {
 			os.Setenv("AWS_WEB_IDENTITY_TOKEN_FILE", val)
+			if val := env.Get("MC_ROLE_ARN", ""); val != "" {
```
Comment (vadmeste) [inline cmd/client-s3.go:223]: If I am not mistaken, a better way is to make mc looks for 'alias', then env vars, then local FS; but this seems to be out of scope of this PR.

**Review by vadmeste — APPROVED** [review]
LGTM

File: cmd/client-s3.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/client-s3.go:223]: Alias is frozen @vadmeste lets not make wide changes for minor requirements, this is good enough for most people.
